tidy3d/plugins/mode/mode_solver.py

```diff
@@ -288,24 +288,38 @@ def _get_epsilon(self, freq: float) -> ArrayComplex4D:
         eps_yy = self.simulation.epsilon_on_grid(self._solver_grid, "Ey", freq)
         eps_zz = self.simulation.epsilon_on_grid(self._solver_grid, "Ez", freq)
 
-        return np.stack((eps_xx, eps_yy, eps_zz), axis=0)
+        eps_xy = self.simulation.epsilon_on_grid(self._solver_grid, "Exy", freq)
+        eps_xz = self.simulation.epsilon_on_grid(self._solver_grid, "Exz", freq)
+        eps_yx = self.simulation.epsilon_on_grid(self._solver_grid, "Eyx", freq)
+        eps_yz = self.simulation.epsilon_on_grid(self._solver_grid, "Eyz", freq)
+        eps_zx = self.simulation.epsilon_on_grid(self._solver_grid, "Ezx", freq)
+        eps_zy = self.simulation.epsilon_on_grid(self._solver_grid, "Ezy", freq)
+
+        return np.stack(
+            (eps_xx, eps_xy, eps_xz, eps_yx, eps_yy, eps_yz, eps_zx, eps_zy, eps_zz), axis=0
+        )
 
     def _solver_eps(self, freq: float) -> ArrayComplex4D:
         """Get the diagonal permittivity in the shape needed to be supplied to the sovler, with the
         normal axis rotated to z."""
 
         # Get diagonal epsilon components in the plane
-        eps_diag = self._get_epsilon(freq)
+        eps_tensor = self._get_epsilon(freq)
 
         # get rid of normal axis
-        eps_diag = np.take(eps_diag, indices=[0], axis=1 + self.normal_axis)
-        eps_diag = np.squeeze(eps_diag, axis=1 + self.normal_axis)
+        eps_tensor = np.take(eps_tensor, indices=[0], axis=1 + self.normal_axis)
+        eps_tensor = np.squeeze(eps_tensor, axis=1 + self.normal_axis)
 
         # swap axes to plane coordinates (normal_axis goes to z)
-        eps_zz, (eps_xx, eps_yy) = self.plane.pop_axis(eps_diag, axis=self.normal_axis)
+        if self.normal_axis == 0:
```

Comment:
Tried to think about it, but couldn't think of anything straightforward. Since we use everywhere in mode solver (`compute_modes()`, `_solver_eps()`, and `_get_epsilon()`) the most common ordering (xx, xy, xz, yx, yy, etc), probably it's ok to leave it as it is? Another thing is that in future we might want to make mode solver more consistent with fdtd scheme for fully anisotropic materials, which would require changing from eps to eps^{-1} and providing more than just 9 tensorial components. So, maybe we can revisit this then?

I did reimplement this particular piece of code using permutations of a tensorial form to make it more clear and more bug-proof for now